presto-main/src/main/java/com/facebook/presto/metadata/Signature.java

```diff
@@ -404,11 +404,16 @@ else if (canCoerce(boundParameters.get(parameter.getBase()), type) && typeParame
         if (!parametersMatched && !typeSignature.getBase().equals(parameter.getBase())) {
             // check for possible coercion
             if (allowCoercion && canCoerce(typeSignature, parameter)) {
+                Type coercedType = typeManager.getType(parseTypeSignature(parameter.getBase()));
```

Comment:
Not sure why you needed this change, but I think you can remove it, if you make the change in FunctionRegistry that I suggested